src/graph/test/IndexTest.cpp

```diff
@@ -687,6 +687,83 @@ TEST_F(IndexTest, EdgeIndexTTL) {
     }
 }
 
+TEST_F(IndexTest, AlterTag) {
+    auto client = gEnv->getClient();
+    ASSERT_NE(nullptr, client);
+    {
+        cpp2::ExecutionResponse resp;
+        std::string query = "CREATE SPACE tag_index_space(partition_num=1, replica_factor=1)";
+        auto code = client->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+
+        query = "USE tag_index_space";
+        code = client->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+
+        query = "CREATE TAG person(name string, age int, gender string, email string)";
+        code = client->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    sleep(FLAGS_heartbeat_interval_secs + 1);
+    // Single Tag Single Field
+    {
+        cpp2::ExecutionResponse resp;
+        std::string query = "CREATE TAG INDEX single_person_index ON person(name)";
+        auto code = client->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    sleep(FLAGS_heartbeat_interval_secs + 1);
+    {
+        cpp2::ExecutionResponse resp;
+        auto query = "INSERT VERTEX person(name, age, gender, email) VALUES "
+                     "100:  (\"Tim\",  18, \"M\", \"tim@ve.com\")";
+        auto code = client->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        auto query = "ALTER TAG person ADD (col1 int)";
```

Comment:
Does it work well when we change the prop name or type? It's better to test it.

Reply:
> Does it work well when we change the prop name or type? It's better to test it.

Don't allow to `alter tag change column` or `alter tag drop column` when the tag attached index. An error message will be return.
If you want to `alter tag change column` or `alter tag drop column` , need drop index firstly.